### PR TITLE
CustomSlider and CustomRangeSlider

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
# Description

Adds new `CustomSlider` and `CustomRangeSlider` components with a truckload of options (check the doc comments and the stories 😉 )

# Screenshots / Videos

https://user-images.githubusercontent.com/77000136/152543486-050b7cfa-857b-4a70-9d63-057ecdbcaeb8.mov

https://user-images.githubusercontent.com/77000136/152543516-5a13b929-8cd1-4db4-ba79-9ad3cc3eeceb.mov

# Linked documentation PR

\-

### CODE REVIEW

File: scripts/components/custom-slider/custom-range-slider.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+ * @param {boolean} [props.dots=false]                                                - If `true` and the `step` is greater than `1`, dots will be rendered on the slider. Won't have any effect if `marks` are set.
+ * @param {function} [props.onBeforeChange]                                           - Function to trigger when `ontouchstart` or `onmousedown` is triggered.
+ * @param {function} [props.onChange]                                                 - Function to trigger when the value of the slider is changing.
+ * @param {function} [props.onAfterChange]                                            - Function to trigger when `ontouchend` or `onmouseup` is triggered.
+ * @param {Number[]} [props.defaultValue=[0,0]]                                       - Initial positions of handles.
+ * @param {Number[]} [props.value]                                                    - Current positions of handles.
+ * @param {Number[]} [props.tabIndex=0]                                               - Sets the `tabIndex` of each slider handle.
+ * @param {string[]} [props.ariaLabelGroupForHandles]                                 - Sets the `arial-label` on each slider handle.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-range-slider.js:39]:
```suggestion
 * @param {string[]} [props.ariaLabelGroupForHandles]                                 - Sets the `aria-label` on each slider handle.
```

File: scripts/components/custom-slider/custom-range-slider.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+ * @param {string[]} [props.ariaLabelledByForHandle]                                  - Sets the `arial-labelledby` on each slider handle.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-range-slider.js:40]:
```suggestion
 * @param {string[]} [props.ariaLabelledByForHandle]                                  - Sets the `aria-labelledby` on each slider handle.
```

File: scripts/components/custom-slider/custom-range-slider.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+ * @param {callback[]} [props.ariaValueTextFormatterForHandle]                        - Sets the `arial-valuetext` on each slider handle. Should be in format `(value) => string`.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-range-slider.js:41]:
```suggestion
 * @param {callback[]} [props.ariaValueTextFormatterForHandle]                        - Sets the `aria-valuetext` on each slider handle. Should be in format `(value) => string`.
```

File: scripts/components/custom-slider/custom-range-slider.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+		disabled,
+		railColor,
+		trackColor,
+		handleColor,
+		leftAddition,
+		rightAddition,
+		activeMarkColor,
+		inactiveMarkColor,
+		activeMarkLabelColor,
+		inactiveMarkLabelColor,
```
Comment (volfkarlo) [inline scripts/components/custom-slider/custom-range-slider.js:89]: happiness chart


How happy people are when reading your PR 😄

File: scripts/index.js [hunk 5]
```diff
@@ -14,6 +14,14 @@ export {
 export { CollapsableComponentUseToggle } from './components/collapsable-component-use-toggle/collapsable-component-use-toggle';
 export { ComponentUseToggle } from './components/component-use-toggle/component-use-toggle';
 export { CustomSelect } from './components/custom-select/custom-select';
+export { 
+	CustomSlider, 
+	CustomSliderStyle
```
Comment (volfkarlo) [inline scripts/index.js:19]:
```suggestion
	CustomSliderStyle,
```

File: scripts/index.js [hunk 6]
```diff
@@ -14,6 +14,14 @@ export {
...
+} from './components/custom-slider/custom-slider';
+export { 
+	CustomRangeSlider,
+	CustomRangeSliderStyle
```
Comment (volfkarlo) [inline scripts/index.js:23]:
```suggestion
	CustomRangeSliderStyle,
```

**Review by volfkarlo — CHANGES_REQUESTED** [review]
Awesome job! 👍  I would say this is really something we need regarding UX 🎉 

Micro changes plus, I haven't seen an import for the component in `scripts/components/index.js`. Please add that part! 😄

File: scripts/components/custom-slider/custom-slider.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+ * @param {function} [props.onBeforeChange]                                 - Function to trigger when `ontouchstart` or `onmousedown` is triggered.
+ * @param {function} [props.onChange]                                       - Function to trigger when the value of the slider is changing.
+ * @param {function} [props.onAfterChange]                                  - Function to trigger when `ontouchend` or `onmouseup` is triggered.
+ * @param {Number} [props.defaultValue=0]                                   - Initial value of the slider.
+ * @param {Number} [props.value]                                            - Current value of the slider.
+ * @param {Number} [props.startPoint=undefined]                             - Determines starting point of the slider track. If `undefined`, value of `min` is used.
+ * @param {Number} [props.tabIndex=0]                                       - Sets the `tabIndex` of the slider handle.
+ * @param {string?} [props.ariaLabelForHandle]                              - Sets the `arial-label` on the slider handle.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-slider.js:36]:
```suggestion
 * @param {string?} [props.ariaLabelForHandle]                              - Sets the `aria-label` on the slider handle.
```

File: scripts/components/custom-slider/custom-slider.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+ * @param {string?} [props.ariaLabelledByForHandle]                         - Sets the `arial-labelledby` on the slider handle.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-slider.js:37]:
```suggestion
 * @param {string?} [props.ariaLabelledByForHandle]                         - Sets the `aria-labelledby` on the slider handle.
```

File: scripts/components/custom-slider/custom-slider.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+ * @param {callback} [props.ariaValueTextFormatterForHandle]                - Sets the `arial-valuetext` on the slider handle. Should be in format `(value) => string`.
```
Comment (iobrado) [inline scripts/components/custom-slider/custom-slider.js:38]:
```suggestion
 * @param {callback} [props.ariaValueTextFormatterForHandle]                - Sets the `aria-valuetext` on the slider handle. Should be in format `(value) => string`.
```

**Review by iobrado — APPROVED** [review]
Very nice! I found few typos in comments and I've added suggestions to fix them. Since these are comment typos I'll approve it as-is.

**Review by iobrado — APPROVED** [review]

**Review by volfkarlo — APPROVED** [review]
👏

**Review by dadadavorin — APPROVED** [review]
Wow 🤩  Really cool!

**Pull request merged**